### PR TITLE
[OS Data Sync] Google fit permissions

### PR DESCRIPTION
**Pull request by brendeny**
This PR integrates Google fit into the app. The onboarding screen should ask for google fit permissions and help the user set permissions to gather the google fit data. There should be a homescreen prompt if the user does not grant permissions.

## Changes
- The changes here implement the react-native-google-fit npm package for google fit auth. Users are able to be added on firebase as a test user and their google fit data can be accessed from their accounts once authed in the onboarding screen. Additionally, auth flows are introduced on the homescreen and settings screen.

## Requirements
N/A

Requires: 
Users have installed Fit App and have Steps and Sleep data

## PR Dependencies
- Depends on #52

## Merged Branches and Other Dependencies
Merged branch(es): https://github.com/bipolarbridges/companion-app/pull/52 

## Subtasks / Progress
- [x] Auth users on onboarding
- [x] Fetch Sleep and Step data
- [x] Card on homescreen if user skips onboarding
- [x] Auth through card on homescreen if user skips onboarding
- [x] Settings page auth

## Preview
This is the google-fit auth flow for onboarding:
https://user-images.githubusercontent.com/46551861/144527268-0a3ecd12-d3e3-4707-9ac0-b9adb6906855.mp4

This is the homescreen google fit auth flow when the user skips auth in the onboarding:
https://user-images.githubusercontent.com/46551861/144527323-dcc6866a-791c-4a2e-ad1f-7b98a8fbc040.mp4

This is the google fit auth through user settings:
https://user-images.githubusercontent.com/46551861/144527347-d38d85d4-fa35-4b99-8291-9d95b8c56ae9.mp4

## Testing
- [x] Manual 
- [ ] Automated

List of added tests:
- N/A

### CODE REVIEW

File: mobile/src/stateMachine/views/healthData/healthScopes.tsx
Comment (brendeny) [inline mobile/src/stateMachine/views/healthData/healthScopes.tsx:95]: @MarkDavies0 there is a small fix needed here where the screen the user ends on after pressing confirm needs to refresh to show the updated auth. Or another option is to send the user to the homescreen after auth success

**Comment by brendeny** [comment]
I have implemented the Google Fit auth and tested it. The permissions are persistent in the users google account once the user grants them, however currently, the user needs to auth every time they close/open the app. This may be because every time the app is refreshed as it is in a development/testing version

File: mobile/src/stateMachine/views/healthData/healthScopes.tsx
Comment (MarkDavies0) [reply mobile/src/stateMachine/views/healthData/healthScopes.tsx:95]: @brendeny I think you need to `await authAndroid()` before calling `AppController.Instance.PromptModal.closeModal();` . Alternatively you could have the view react to the new auth status and re-render itself

File: mobile/src/constants/images/app.ts
Comment (MarkDavies0) [inline mobile/src/constants/images/app.ts:52]: Naming here is off, the actual file google_fit_health_perm.png in unused while android_check.png is being imported as GoogleFitHealthPermPNG

File: mobile/src/helpers/health.ts
Comment (MarkDavies0) [inline mobile/src/helpers/health.ts:48]: removed commented code throughout

File: mobile/src/helpers/health.ts
Comment (MarkDavies0) [inline mobile/src/helpers/health.ts:79]: make declaration inline with assignment
`let stepsData = await fetchStepsData(opt);`

File: mobile/src/helpers/health.ts
Comment (MarkDavies0) [inline mobile/src/helpers/health.ts:16]: prefer let or const over var, probably const here

File: mobile/src/helpers/health.ts
Comment (MarkDavies0) [inline mobile/src/helpers/health.ts:33]: This for loop seems like overkill, maybe try Array.find

File: mobile/src/helpers/health.ts
Comment (MarkDavies0) [inline mobile/src/helpers/health.ts:87]: There are a lot of logger and console.log statements in this file, remove all except the most useful for future devs

File: mobile/src/helpers/health.ts
Comment (MarkDavies0) [inline mobile/src/helpers/health.ts:14]: Move this into the scope where it is used

File: mobile/src/helpers/health.ts
Comment (MarkDavies0) [reply mobile/src/helpers/health.ts:16]: `lastWeekDate` should be const

File: mobile/src/helpers/health.ts
Comment (MarkDavies0) [inline mobile/src/helpers/health.ts:46]: `midnight` is used as a date object so I would declare it as an actual date and save on Date.parse further down

File: mobile/src/helpers/health.ts
Comment (MarkDavies0) [inline mobile/src/helpers/health.ts:52]: this is for loop is quite difficult to interpret, any way to simplify? A comment describing what the loop is doing would be helpful

File: mobile/src/stateMachine/scenario.ts
Comment (MarkDavies0) [inline mobile/src/stateMachine/scenario.ts:189]: remove commented line

File: mobile/src/stateMachine/views/main/home.tsx
Comment (MarkDavies0) [inline mobile/src/stateMachine/views/main/home.tsx:305]: remove

File: mobile/src/stateMachine/views/main/home.tsx
Comment (MarkDavies0) [inline mobile/src/stateMachine/views/main/home.tsx:62]: uncomment right before merge

**Review by MarkDavies0 — CHANGES_REQUESTED** [review]
Some changes requested but looking good. Also make sure to delete the template parts of the PR template

File: mobile/src/helpers/health.ts
Comment (brendeny) [reply mobile/src/helpers/health.ts:46]: I had tried to set midnight as a date object, however in order to make the comparison between re[i].startDate and midnight, we need it as a string.

let midnight = new Date();
midnight.setHours(0,0,0,0);

File: mobile/src/helpers/health.ts
Comment (brendeny) [reply mobile/src/helpers/health.ts:52]: left comment

**Comment by brendeny** [comment]
@MarkDavies0 thanks for the review! Made the changes requested. Feel free to reach out to me if you have any further questions. Cheers

**Comment by MarkDavies0** [comment]
@wynnset Tested on iOS and Android devices, needs final review before merge

File: mobile/src/stateMachine/scenario.ts
Comment (wynnset) [inline mobile/src/stateMachine/scenario.ts:195]: How come we are no longer checking `VM.hasHealthPermissions`?

**Review by wynnset — COMMENTED** [review]
Hey @MarkDavies0 it seems like the file "mobile/src/assets/images/google_fit_health_perm.png" is not used anywhere. Or am I mistaken? Also, please check the one comment below.

File: mobile/src/stateMachine/scenario.ts
Comment (MarkDavies0) [reply mobile/src/stateMachine/scenario.ts:195]: The new methods for android and iOS are used instead

**Review by wynnset — APPROVED** [review]

**Pull request merged**
